import_activity_events.py

```diff
@@ -111,27 +117,41 @@
       ua_os
     )
     SELECT
-      'epoch'::TIMESTAMP + timestamp * '1 second'::INTERVAL,
+      ts,
       uid,
       type,
       device_id,
       service,
       ua_browser,
       ua_version,
       ua_os
-    FROM temporary_raw_activity_data;
+    FROM (
+      SELECT
+        *,
+        'epoch'::TIMESTAMP + timestamp * '1 second'::INTERVAL AS ts,
+        STRTOL(SUBSTRING(uid FROM 0 FOR 8), 16) % 100 AS cohort
+      FROM temporary_raw_activity_data
+    )
+    WHERE cohort <= {percent}
+      AND ts::DATE >= '{day}'::DATE - '{months} months'::INTERVAL;
+"""
+
+Q_DELETE_EVENTS = """
+    DELETE FROM activity_events{suffix}
+    WHERE timestamp::DATE < '{day}'::DATE - '{months} months'::INTERVAL;
```

Comment:
You could do a sub-select here like `timestamp::DATE < (SELECT MAX(timestamp::DATE) FROM activity_events{suffix}) - '{months} months'::INTERVAL` if you didn't want to have to pas in `{day}` as a parameter.  It's not obvious to me how strictly we're guaranteed that the day passed in here is the largest day in the table.